### PR TITLE
wth - step 3 additional details link bug

### PR DESCRIPTION
**Pull request by wtholt**
Reordered logic to include line_item_id param, which fixes this issue.
Also, noticed logic errors in Dashboard. [#132778407]

### CODE REVIEW

File: app/views/additional_details/_dashboard_complete_additional_details.html.haml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 - if service_request.additional_detail_services.present?
   - service_request.line_items.each do |line_item|
-    - unless line_item.submissions.present?
-      %div.alert.alert-warning
-        %h5.text-center
-          = link_to t("additional_details.dashboard_complete_details", service_name: Service.find(line_item.service_id).name),
-            new_service_additional_details_submission_path(line_item.service_id, protocol_id: @protocol.id, line_item_id: line_item.id), remote: true
+    - service = Service.find(line_item.service_id)
+    - if service.questionnaires.active.present?
+      - unless line_item.submissions.present?
```
Comment (jayhardee9) [inline app/views/additional_details/_dashboard_complete_additional_details.html.haml]: I think the expression in the `unless` should be tacked on to the `if` clause on line 4.

File: app/views/additional_details/_document_management_submissions.html.haml [hunk 2]
```diff
@@ -15,30 +15,33 @@
...
+                = service.identity_submission(current_identity).present? ? "True" : "False"
+            %td
+              .text-center
+                - if service.identity_submission(current_identity).present?
+                  = link_to edit_service_additional_details_submission_path(service, service.identity_submission(current_identity).first, sr_id: @service_request.id), remote: true, class: 'btn btn-warning' do
+                    %span.glyphicon.glyphicon-edit
+            %td
+              .text-center
+                - if service.identity_submission(current_identity).present?
+                  = link_to "javascript:void(0);", class: 'btn btn-danger delete-submission', data: { id: "#{service.identity_submission(current_identity).first.id}", service_id: "#{service.id}", sr_id: "#{@service_request.id}", protocol_id: "#{@service_request.protocol_id}", line_item_id: "#{line_item.id}" } do
```
Comment (jayhardee9) [inline app/views/additional_details/_document_management_submissions.html.haml]: Also, is it necessary to string-interpolate all these data attributes? Just curious.

Comment (wtholt) [inline app/views/additional_details/_document_management_submissions.html.haml]: Apparently, not. Thanks man. Reference - http://stackoverflow.com/questions/31061716/interpolate-variables-into-data-attributes

File: app/views/additional_details/_dashboard_complete_additional_details.html.haml (hunk 1 above)
Comment (wtholt) [inline app/views/additional_details/_dashboard_complete_additional_details.html.haml]: Agreed, thanks

**Pull request merged**